client/test/e2e/shared/components.js

```diff
@@ -134,6 +134,15 @@ exports.currencyInput = {
   },
 
   /**
+  *set the value of the component by his id to avoid conflict
+  */
+  setById : function setById (id, value){
```

Comment:
Instead of creating a new method `setById()`, why not just update the old method?  I would do it like this:

First, change the select from the `<input ...>` to the [top level element](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/partials/templates/bhCurrencyInput.tmpl.html#L1) in the bhCurrencyInput.  Something like:
```html
<div class="form-group" data-bh-currency-input>
 <!-- code ... -->
</div>
```
Then, in the JavasScript, you can use a root element to select the correct currencyInput each time.
```js
set : function set(value, id) {

  // it might be clearer to do this in two steps.
 var root = element(id ? by.id(id) : by.css(this.selector));

  var elm  = root.element(by.model('$ctrl.model');
  elm.sendKeys(value);
}
```